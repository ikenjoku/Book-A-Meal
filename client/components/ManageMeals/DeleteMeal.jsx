import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { removeAMeal } from "../../actions/mealActions";

export const DeleteMeal = ({ meal, closeModal, removeAMeal }) => {

  return (
    <div className='confirm-modal'>
      <h3>Are you sure you want to delete {meal.name}?</h3>
      <div>
        <button id='confirm-delete-btn' onClick={() => {
          removeAMeal(meal.id);
          closeModal();
          }}>Yes</button>
        <button onClick={closeModal}>No</button>
      </div>
    </div>
  );
};

DeleteMeal.propTypes = {
  meal: PropTypes.object.isRequired,
  closeModal: PropTypes.func.isRequired,
  removeAMeal: PropTypes.func.isRequired,
}

export default connect(undefined, { removeAMeal })(DeleteMeal);