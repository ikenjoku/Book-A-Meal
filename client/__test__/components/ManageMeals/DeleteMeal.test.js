import React from 'react';
import { shallow } from "enzyme";
import { DeleteMeal } from '../../../components/ManageMeals/DeleteMeal.jsx'


let wrapper;

describe.only('AddMeal Component', () => {
  beforeEach(() => {
    wrapper = shallow(<DeleteMeal />);
  });

  test('should render the DeleteMeal correctly', () => {
    wrapper.find('#confirm-delete-btn').simulate('click')
    expect(wrapper).toMatchSnapshot();
  });
});