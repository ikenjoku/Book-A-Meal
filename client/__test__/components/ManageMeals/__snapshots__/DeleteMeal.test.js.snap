// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteMeal Component should handle delete button 1`] = `
<div
  className="confirm-modal"
>
  <h3>
    Are you sure you want to delete 
    Nigerian Jollof
    ?
  </h3>
  <div>
    <button
      id="confirm-delete-btn"
      onClick={[Function]}
    >
      Yes
    </button>
    <button
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
    >
      No
    </button>
  </div>
</div>
`;

exports[`DeleteMeal Component should render the DeleteMeal correctly 1`] = `
<div
  className="confirm-modal"
>
  <h3>
    Are you sure you want to delete 
    Nigerian Jollof
    ?
  </h3>
  <div>
    <button
      id="confirm-delete-btn"
      onClick={[Function]}
    >
      Yes
    </button>
    <button
      onClick={[MockFunction]}
    >
      No
    </button>
  </div>
</div>
`;
